### DISCUSSION TITLE
Diaspora* social link

### DISCUSSION THREAD

**Discussion by stevolucidity** [HEIGE-PCloud/DoIt#388] [Q&A]
Can someone please share how to include a Diaspora* link along with Twitter, GitHub etc?
I don't think it's included in the current social links.

Please provide an example config.toml section?

Thank you.

**Comment by eyllanesc** [comment]
I could create a PR where that social network is added but I need the following: 

- [x] Prefix: For example, in the case of github it is `https://github.com/` since the link of your user is `https://github.com/<username>`, what is the diaspora? What is the link to your account?
- [x] Title: `diaspora*`
- [x] Icon: `Simpleicons: diaspora`

**Comment by stevolucidity** [comment]
Hi, Edwin. Thanks for the input.

The prefix would depend on which 'pod' the user is signed up to.
[Diaspora*](https://diasporafoundation.org) is a de-centralised social platform not beholden to any one organisation or company, but federated in the same way as Mastodon and Hubzilla.
I host my own diaspora* pod. The link to find me [is here](https://deafaspora.simons.earth/people/7506c78027ba013af1aa0242ac140004).

I'm wondering if it were possible to do this via the config.toml?

I tried using the following from something I found in a thread a couple of weeks ago.
```
        [languages.en.params.social.Diaspora]
          weight = 0
          id = "7506c78027ba013af1aa0242ac140004"
          prefix = "https://deafaspora.simons.earth/people/"
          title = "Diaspora"
```
but Hugo throws the error:
```
Change of config file detected, rebuilding site.
2021-12-10 22:22:17.139 +0000
ERROR 2021/12/10 22:22:17 Failed to reload config:
ERROR 2021/12/10 22:22:17 "/Users/steve/blog.simons.earth/config.toml:1:1": unmarshal failed: unhandled kv part: string
Rebuilt in 0 ms
```

**Reply by eyllanesc** [reply]
I have already created the PR #393  that adds diaspora, you just have to change:

https://github.com/HEIGE-PCloud/DoIt/blob/2b2a7895259fb894f6c392951a3f1bf38b42fd01/exampleSite/config.toml#L268-L270

to
```toml
        [languages.en.params.social.Diaspora]
          id = "7506c78027ba013af1aa0242ac140004"
          prefix = "https://deafaspora.simons.earth/people/"
```

in config.toml

**Reply by eyllanesc** [reply]
![Screenshot_20211214_095631](https://user-images.githubusercontent.com/4896200/146022587-f362cc4f-1313-4ff1-a26d-6a3c2cef9a0b.png)

**Reply by stevolucidity** [reply]
Thank you for the PR. I appreciate your efforts.

Unfortunately after making the few changes to social.yml and config.toml shown in the diffs (I couldn't figure out how to merge your branch), I still get the same error I was encountering before.

When the PR gets merged upstream I'll do a cleanup and start from scratch.

Kind regards.

**Reply by HEIGE-PCloud** [reply]
@evoLucidity The PR has been merged to the main branch, check it out! :)

**Reply by stevolucidity** [reply]
That's great news but I can't get it to work at my end.

With the following in config.toml:
```
        RSS = false
#        [languages.en.params.social.Mastodon]
#          id = "@xxxx"
#          prefix = "https://mastodon.technology/"
        [languages.en.params.social.Diaspora]
          id = "7506c78027ba013af1aa0242ac140004"
          prefix = "https://deafaspora.simons.earth/people/"
      # Sponsor config
      [languages.en.params.sponsor]
        enable = false
```

I get 
```
steve@Steves-MBP blog.simons.earth % hugo server -D
Error: "/Users/steve/blog.simons.earth/config.toml:1:1": unmarshal failed: unhandled kv part: string
```
...and the page won't render.

Without it:
```
        RSS = false
#        [languages.en.params.social.Mastodon]
#          id = "@xxxx"
#          prefix = "https://mastodon.technology/"
#        [languages.en.params.social.Diaspora]
#          id = "7506c78027ba013af1aa0242ac140004"
#          prefix = "https://deafaspora.simons.earth/people/"
      # Sponsor config
      [languages.en.params.sponsor]
        enable = false
```
it works fine (minus the diaspora link).
![Image 14-12-2021 at 17 02](https://user-images.githubusercontent.com/44602820/146044938-742aad1f-eafb-4f87-bf9a-d162880f8a9c.jpg)

I don't know enough (anything) about Hugo to investigate further. Could you possibly suggest a way of making the output offer more of a clue (--verbose and --debug don't produce anything more).

**Reply by HEIGE-PCloud** [reply]
Cannot reproduce... It seems that the config.toml cannot be parsed correctly after this is enabled. Can you share a minimal config.toml that reproduces the problem?

**Reply by stevolucidity** [reply]
I figured out that the `Diaspora = ""` is _mandatory_ after the social.Diaspora definition.
I had it before the definition.
It's now working as expected.

Apologies both, and thank you very much for the speedy turn-around.
